src/Stratis.Bitcoin.IntegrationTests/BlockStore/RetrieveFromBlockStoreSpecification.cs

```diff
@@ -0,0 +1,48 @@
+﻿using Stratis.Bitcoin.IntegrationTests.TestFramework;
+using Xunit;
+
+// Disable warnings about "this" qualifier to make the Specification more readable
+// ReSharper disable ArrangeThisQualifier
+
+namespace Stratis.Bitcoin.IntegrationTests.BlockStore
+{
+    public partial class RetrieveFromBlockStoreSpecification : BddSpecification
+    {
+        [Fact]
+        public void A_block_can_be_retrieved_by_its_uint256_identifier()
+        {
+            Given(a_pow_node_running);
+            And(a_miner_validating_blocks);
+            And(some_real_blocks_with_a_uint256_identifier);
+            And(a_made_up_block_id);
+            And(the_node_is_synced);
+
+            When(trying_to_retrieve_the_blocks_from_the_blockstore);
+
+            Then(real_blocks_should_be_retrieved);
+            Then(made_up_blocks_should_not_be_retrieved);
+        }
+
+        [Fact]
+        public void A_transaction_can_be_retrieved_by_its_uint256_identifier()
```

Comment:
Hi, could the real and made up blocks be separate tests ? The unit test name `A_block_can_be_retrieved_by_its_uint256_identifier()` suggests you are testing for a good block only.  

Or change the name to `A_block_can_be_retrieved_by_its_uint256_identifier_and_a_made_up_block_rejected()` ?

Reply:
OK sorry, I think I will rename unless someone insists on separating the tests, I think the set up of these tests is so long that in that case grouping a trivial test with a normal one is valid... Thanks!

Reply:
Yes on thought so to.